src/map.rs

```diff
@@ -117,16 +126,88 @@ impl GameMap {
         }
     }
 
+    pub fn valid_tile_pos(&self, x: isize, y: isize) -> Option<(usize, usize)> {
+        if x > 0 && y > 0 && x < self.xsize as isize && y < self.ysize as isize {
```

Comment:
we should expose `xsize` and `ysize` gobal and then not have this function be `(&self, ...`